contracts/epochs/FlowEpoch.cdc

```diff
@@ -1296,6 +1296,41 @@ access(all) contract FlowEpoch {
                 ?? 0.0
     }
 
+    /// Checks if the current epoch is in a phase transition
+    /// so that other system contracts can skip computation-heavy operations
+    /// during the phase transition.
+    access(all) fun isPhaseTransition(): Bool {
```

Comment:
It makes me nervous that this is duplicating the state machine transition logic from `advanceBlock`. Any future changes would need to be carefully applied to both functions. 

I know it's hard to refactor deployed contracts, so maybe there just isn't a good way to do this, but I'm wondering if there is a way to use the same logic here and in `advanceBlock`. Maybe this function could return an enum which maps to a possible state machine transition in `advanceBlock`, like` Enum{``PayRewards``, ``StakingToSetup``, ``SetupToCommitted``, ``EndEpoch``, ``Noop``}`. Then `advanceBlock` could switch on the returned enum, and callers interested in whether any expensive operation is occurring could check `isPhaseOperation != Noop`.